blocks/filter-group/filter-group.js

```diff
@@ -1,42 +1,193 @@
+        if (mockupTagsAppliedToCards.length < i) {
+            card.dataset.tags = 'Design Stategy';
+            return;
+        }
+        card.dataset.tags = mockupTagsAppliedToCards[i];
+    });
+}
+
+/**
+ * TO-DO: work with real Ideas page content. This function is currently mockup/placeholder functionality
+ * and will likely need to be moved to the ideas/stories feed block.
+ * 
+ * Update all articles displayed on the page with ones that have tags
+ * that match the currently selected filters.
+ */
+const refreshArticleContent = (selectedFilters = []) => {
+    // TO-DO: remove this when ideas page content and functionality is finalized.
+    applyTemporaryMockupTags();
+
+    // Show/hide article cards based on selected tags.
+    document.querySelectorAll('.card')?.forEach(article => {
+        // "All" reset. Display all articles.
+        if (selectedFilters.length == 0){
+            article.style.display = '';
+            return;
+        }
+        
+        // Show/hide based on data-tags applied to the article.
+        if (article.dataset?.tags) {
+            const tagsArray = article.dataset.tags.split(',');
+            const hasTag = selectedFilters.some(tag => tagsArray.includes(tag));
+            article.style.display = hasTag ? '' : 'none';
+        }
+    });
+};
+
+/**
+ * Handle click on a filter button from the parent container. Events are bubbled
+ * up from the button. Triggers functionality to update filtered content and
+ * selected state.
+ * @param {PointerEvent} event
+ */
+const handleFilterClick = function (event) {
+    const filterGroup = this;
+    const clickedButton = event.target.closest('.filter-group__button');
+    if (!clickedButton) return;
+
+    // State of button and whether it's the special "All" filter reset.
+    const isSelected = clickedButton.getAttribute('aria-pressed') == 'true';
+    const isAllFilter = clickedButton.matches(':first-of-type');
+
+    if (isAllFilter) {
+        // "All" filter -----
+        // Does nothing if already selected.
+        if (isSelected) return;
+        // Set all other filters as unselected when "All" is selected.
+        filterGroup.querySelectorAll('.filter-group__button').forEach((btn) => {
+            updateFilter(btn, false);
+        });
+        updateFilter(clickedButton, true);
+    } else {
+        // Regular filters -----
+        // Set filter as selected, and deselect "All" if it was selected.
+        const allButton = filterGroup.querySelector('.filter-group__button:first-of-type');
+        updateFilter(clickedButton, !isSelected);
+        
+        // Update state of "All" filter:
+        // - If at least one filter is selected, "All" gets deselected.
+        // - In the case that the only selected filter was just deselected, set "All" back to selected.
+        const hasAnySelectedFilter = document.querySelector('.filter-group__button:not(:first-of-type)[aria-pressed="true"]') != null;
```

Comment:
we're a little inconsistant in the queries in this file, too, for example, this one is querying `document` and the others are more scoped to `filterGroup`. Can we scope them all to `filterGroup`?

Reply:
Good catch. This is updated along with the refactor.